hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/FileSystemContractBaseTest.java

```diff
@@ -526,6 +526,18 @@ public void testRenameDirectoryMoveToExistingDirectory() throws Exception {
         fs.exists(path(dst + "/subdir/file2")));
   }
 
+  @Test
+  public void testStatisticsDataReferenceCleanerClassLoader() {
+    // initialize the static thread
+    new FileSystem.Statistics("test");
+    Thread.getAllStackTraces().keySet().stream()
+        .filter(t -> t.getName().contains("StatisticsDataReferenceCleaner")).findFirst()
+        .ifPresent(t -> {
+          ClassLoader classLoader = t.getContextClassLoader();
+          assertNull(classLoader);
+        });
```

Comment:
nice.
Can you put it in `TestFileSystemStorageStatistics` as I think it is a better home. Putting it in this test suite will actually rerun it multiple times: once per subclass

Reply:
done